inductiva/resources/machines.py

```diff
@@ -32,20 +28,28 @@ def __init__(
             disk_size_gb: The size of the disk in GB, recommended min. is 20 GB.
             name: The name to assign to the machine group.
         """
+        #TODO: Check if machine type is valid.
         self.machine_type = machine_type
         self.num_machines = num_machines
         self.spot = spot
         self.disk_size_gb = disk_size_gb
         self.name = self._generate_instance_name()
-        self.machine_group_id = resources.create_machine_group_id()
         self.zone = zone
-        self.api_config = api.validate_api_key(inductiva.api_key)
+        self.estimated_cost = -1  # The negative implies that the cost is unknown.
+
+        # Validate the specific user configurations/privileges from the API.
```

Comment:
It's still not clear what this does.
If we are validating the privileges, then there should be some exception thrown if the user does not meet the privileges required for this machine group, right? If there is a missing feature please write a TODO. 
Also, what is this api_config? It's not clear to me...

Reply:
The api_config carries the API configuration from the client to the backend. In particular, the api_url and api_key.

All the required authentications of the privileges are validated in the backend and the exception is thrown from there. Notice, at the moment we are only verifying if the user is admin or not.

Reply:
please add comments about exactly that. 
Also, I don't think this is the proper moment to do this and I think there are two different things here mixed. 

1) do I have access to API

2) can I launch this sort of resources. 

These are two different things.
It's ok for now, but we need to think about this later.

Reply:
This doesn't validate user privileges yet, it only validates the api key and makes sure that it exists in out database. And only when we send the start() request to the API that's when we do all the checks before launching the machine I will open a PR with that very soon.

Reply:
Indeed they are, so we can have two different checks for privileges. In all calls to the API, we need to perform the above validation, the second point will only occur for the resources.